mitm_service/protocol_filters.py

```diff
@@ -22,34 +22,24 @@ def generate_ip_redirect_rule(target_ip, redirect_ip):
     target_ip = inet_aton(target_ip)
     redirect_ip = inet_aton(redirect_ip)
 
-    def disrupt_ip_traffic(ip_header, ip_payload):
-        if not isinstance(ip_header, IpHeader):
+    def disrupt_ip_traffic(scapy_pkt):
+        if IP not in scapy_pkt:
             # not an ip packet so ignore
-            return ip_header, ip_payload
+            return scapy_pkt
 
-        if ip_header.dst_ip != target_ip and ip_header.src_ip != redirect_ip:
+        ip_header = scapy_pkt[IP]
+        if ip_header.dst != target_ip and ip_header.src != redirect_ip:
             return ip_header, ip_payload
 
-        if ip_header.dst_ip == target_ip:
+        if ip_header.dst == target_ip:
             print('Redirecting packet meant for {} to {}'.format(ip_header.dst_ip_str, inet_ntoa(redirect_ip)))
-            ip_header.dst_ip = redirect_ip
+            ip_header.dst = redirect_ip
 
-        elif ip_header.src_ip == redirect_ip:
+        elif ip_header.src == redirect_ip:
             print('Spoofing packet origin from {} to {}'.format(ip_header.src_ip_str, inet_ntoa(target_ip)))
-            ip_header.src_ip = target_ip
-
-        # adjust checksums, if they exist
-        if ip_header.proto == IPPROTO_TCP:
-            l4_header, l4_header_len = TcpHeader.parse_raw_header(ip_payload)
-        elif ip_header.proto == IPPROTO_UDP:
-            l4_header, l4_header_len = UdpHeader.parse_raw_header(ip_payload)
-        else:
-            return ip_header, ip_payload
-        l4_payload = ip_payload[l4_header_len:]
-        l4_header.checksum = 0
-        ip_payload = l4_header.get_raw_header() + l4_payload
+            ip_header.src = target_ip
 
-        return ip_header, ip_payload
+        return scapy_pkt
```

Comment:
although changes will persist w/o, i think explicit set of scapy_pakt[IP] = ip_header before return of scapy_pakt makes it much more readable

Reply:
yeah thats just wrong no

Reply:
lol